Sources/XMTPiOS/Group.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  Group.swift
+//
+//
+//  Created by Pat Nakajima on 2/1/24.
+//
+
+import Foundation
+import LibXMTP
+
+public struct Group: Identifiable, Equatable, Hashable {
+	var ffiGroup: FfiGroup
+	var client: Client
+
+	public struct Member {
+		var ffiGroupMember: FfiGroupMember
+
+		public var accountAddress: String {
+			ffiGroupMember.accountAddress
+		}
+	}
+
+	public var id: Data {
+		ffiGroup.id()
+	}
+
+	public static func == (lhs: Group, rhs: Group) -> Bool {
+		lhs.id == rhs.id
+	}
+
+	public func hash(into hasher: inout Hasher) {
+		id.hash(into: &hasher)
+	}
+
+	public func members() async throws -> [Member] {
+		_ = try await ffiGroup.sync()
+		return try ffiGroup.listMembers().map(\.fromFFI)
+	}
+
+	public var cachedMembers: [Member] {
+		do {
+			return try ffiGroup.listMembers().map(\.fromFFI)
+		} catch {
+			return []
+		}
+	}
```

Comment:
I ended up just make a list of member addresses instead of a list of members what else is on the member is useful to integrators?

Reply:
For now I think member addresses is the only interesting thing, tho I could see that changing if we ever introduce some notion of nicknames or maybe who added whom. I was mostly just mirror LibXMTP here but if we just want addresses for consistency between ios and android I can make the change.

Reply:
Since memberAdd and memberRemove take an address and not a member object. I think right now it makes more sense to just have a list of addresses. In the future if members become more advanced than we can handle that with a new class.